src/lib/functions/runtimes/js/index.ts

```diff
@@ -118,6 +130,7 @@ export const invokeFunctionDirectly = async ({ context, event, func, timeout })
     lambdaPath,
     timeoutMs: timeout * SECONDS_TO_MILLISECONDS,
     verboseLevel: 3,
+    esm: lambdaPath.endsWith('.mjs'),
```

Comment:
We're already running ESM functions locally today, so I'm not sure I follow why we're adding this now. What does it do exactly?

Reply:
We respect the module format specified by the file extension and the `module` property. So we're keeping ESM files as ESM, at least in production. I'd be surprised if this wasn't the case locally. The CLI itself is ESM, so by importing `lambda-local` as ESM it should be able to import both ESM and CJS files without a dynamic import?

Reply:
But lambda-local is a CJS module, and if we don't set the `esm` property then it will always `require` the file at `lambdaPath`. See here: https://github.com/ashiina/lambda-local/blob/8914e6804533450fa68c56fe6c34858b645735d0/src/lambdalocal.ts#L337-L344

I'm not sure why this hasn't been a problem before, but if I remove this line locally I get:

![Screenshot 2024-05-07 at 13 19 48](https://github.com/netlify/cli/assets/14912729/62a820f1-c98d-4032-97a8-0f9321c94eb3)